terraform/modules/s3/s3.tf

```
@@ -40,3 +40,41 @@ resource "aws_s3_bucket" "s3-nwp-bucket" {
 data "aws_s3_bucket" "s3-ml" {
   bucket = "nowcasting-ml-models-${var.environment}"
 }
+
+# Creates S3 bucket for satellite data
+
+# make sure all public access is blocked
+# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_public_access_block
+resource "aws_s3_bucket_public_access_block" "s3-sat-block-public-access" {
+  bucket = aws_s3_bucket.s3-sat-bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  restrict_public_buckets = true
+  ignore_public_acls      = true
+
+}
+
+# Documentation: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket
+resource "aws_s3_bucket" "s3-sat-bucket" {
+  bucket = "nowcasting-nwp-${var.environment}"
```

Probably should change the name here as well, I think

```suggestion
  bucket = "nowcasting-sat-${var.environment}"
```